data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py

```diff
@@ -0,0 +1,268 @@
+from pathlib import Path
+import math
+import pandas as pd
+
+from data_pipeline.etl.base import ExtractTransformLoad
+from data_pipeline.score import field_names
+from data_pipeline.utils import get_module_logger, unzip_file_from_url
+
+logger = get_module_logger(__name__)
+
+
+class EPARSEISCOREETL(ExtractTransformLoad):
+    """Class for 2019 Census Tract Aggregated micro-data
+
+    Data source overview: Page 20 in this document:
+    https://www.epa.gov/sites/default/files/2017-01/documents/rsei-documentation-geographic-microdata-v235.pdf
+
+    Disaggregated and aggregated datasets for 2019 is documented here:
+    https://github.com/usds/justice40-tool/issues/1070#issuecomment-1005604014
+
+    """
+
+    def __init__(self):
+        self.AGGREGATED_RSEI_SCORE_FILE_URL = "http://abt-rsei.s3.amazonaws.com/microdata2019/census_agg/CensusMicroTracts2019_2019_aggregated.zip"
+
+        self.OUTPUT_PATH: Path = (
+            self.DATA_PATH / "dataset" / "epa_rsei_aggregated_microdata"
+        )
+
+        self.TRACT_INPUT_COLUMN_NAME = "GEOID10"
+        self.NUMBER_FACILITIES_INPUT_FIELD = "NUMFACS"
+        self.NUMBER_RELEASES_INPUT_FIELD = "NUMRELEASES"
+        self.NUMBER_CHEMICALS_INPUT_FIELD = "NUMCHEMS"
+        self.AVERAGE_TOXICITY_INPUT_FIELD = "TOXCONC"
+        self.SCORE_INPUT_FIELD = "SCORE"
+        self.POPULATION_INPUT_FIELD = "POP"
+        self.CSCORE_INPUT_FIELD = "CSCORE"
+        self.NCSCORE_INPUT_FIELD = "NSCORE"
+
+        # References to the columns that will be output
+        self.COLUMNS_TO_KEEP = [
+            self.GEOID_TRACT_FIELD_NAME,
+            field_names.EPA_RSEI_NUMBER_FACILITIES_OUTPUT_FIELD,
+            field_names.EPA_RSEI_NUMBER_RELEASES_OUTPUT_FIELD,
+            field_names.EPA_RSEI_NUMBER_CHEMICALS_OUTPUT_FIELD,
+            field_names.EPA_RSEI_AVERAGE_TOXICITY_OUTPUT_FIELD,
+            field_names.EPA_RSEI_SCORE_OUTPUT_FIELD,
+            field_names.EPA_RSEI_CSCORE_OUTPUT_FIELD,
+            field_names.EPA_RSEI_NCSCORE_OUTPUT_FIELD,
+            field_names.EPA_RSEI_POPULATION_OUTPUT_FIELD,
+            field_names.EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_NCSCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_CSCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD,
+            field_names.EPA_RSEI_SCORE_THRESHOLD_FIELD,
+        ]
+
+        self.df: pd.DataFrame
+
+    def _custom_percentile(self, data, percentile):
```

Comment:
I'm not sure I understand why you needed to create this custom function and you didn't use something like `np.nanquantile(a= data, q= percentile)`?

Reply:
Please see my comments below and in the PR introduction. I hope that is helpful.

Reply:
Not pertinent anymore as there is no expressed desire to include stratification or examining different portions of the distribution of scores in conjunction with other datasets.